src/parser.rs

```diff
@@ -38,15 +39,21 @@ impl<'a> Packet<'a> {
             nameservers.push(try!(parse_record(data, &mut offset)));
         }
         let mut additional = Vec::with_capacity(header.additional as usize);
+        let mut opt = None;
         for _ in 0..header.additional {
-            additional.push(try!(parse_record(data, &mut offset)));
+            if data[offset..offset+3] == OPT_RR_START {
+                opt = parse_opt_record(data, &mut offset).ok();
```

Comment:
Yes, error checking still needs work. I'll fix those up.

I'm not sure about taking this record out of the additional records either. Both have downsides. For example leaving the OPT record as an existing ResourceRecord would lose type safety for the class field as it's used as max UDP packet size. Would working around this make the interface to the additional records more complicated? How would an alternative look? Making ResourceRecord an Enum?